app/src/main/java/com/conupods/OutdoorMaps/BuildingDataMap.java

```diff
@@ -52,8 +52,77 @@ private void parseBuildingData() {
             jsonReader.beginArray();
             while (jsonReader.hasNext()) {
                 Building building = readBuildingJsonObject(jsonReader);
+                List<String> classRooms = null;
+                String campus = null;
+                String code = null;
+                String name = null;
+                String longName = null;
+                String address = null;
+                double latitude = Double.NaN;
+                double longitude = Double.NaN;
+                double overlayLatitude = Double.NaN;
+                double overlayLongitude = Double.NaN;
 
                 mData.put(building.getLatLng(), building);
+                jsonReader.beginObject();
+
+                while (jsonReader.hasNext()) {
+                    String buildingData = jsonReader.nextName();
+                    switch (buildingData) {
+                        case "Classrooms":
+                            jsonReader.beginArray();
+                            classRooms = fillBuildingWithClassrooms(jsonReader, buildingData);
+                            jsonReader.endArray();
+                            break;
+                        case "Campus":
+                            campus = jsonReader.nextString();
+                            break;
+                        case "Building":
+                            code = jsonReader.nextString();
+                            break;
+                        case "BuildingName":
+                            name = jsonReader.nextString();
+                            break;
+                        case "Building Long Name":
+                            longName = jsonReader.nextString();
+                            break;
+                        case "Address":
+                            address = jsonReader.nextString();
+                            break;
+                        case "Latitude":
+                            latitude = jsonReader.nextDouble();
+                            break;
+                        case "Longitude":
+                            longitude = jsonReader.nextDouble();
+                            break;
+                        case "Overlay Latitude":
+                            overlayLatitude = jsonReader.nextDouble();
+                            break;
+                        case "Overlay Longitude":
+                            overlayLongitude = jsonReader.nextDouble();
+                            break;
+                        default:
+                            jsonReader.skipValue();
+                            break;
+                    }
+                }
+                jsonReader.endObject();
+
+                // Save the metadata to the map object of the singleton
+                LatLng latLng = new LatLng(latitude, longitude);
```

Comment:
This is the old reader function, there should be 2 new functions `readBuildingJsonObject()` and `readClassroomJsonArray()`
I suggest u copy paste this file from the master and re-add ur overlayLong and overlayLat in the correct places